src/MonitorProfiler/MainProfiler/ExceptionTracker.cpp

```diff
@@ -0,0 +1,392 @@
+    - Phase 3: Unwinding.
+      The following are invoked for each frame starting at the leaf frame (the frame from which the exception is thrown):
+      - ExceptionUnwindFunctionEnter: Invoked when beginning to unwind a frame.
+      If a matching handler was found, the unwind phase ends after the invocation of
+      the ExceptionUnwindFunctionEnter callback for the frame that contains the matching handler.
+      - ExceptionUnwindFunctionLeave: Invoked when finishing the unwind of a frame.
+      If all of the frames are unwound to completion, the process is terminated.
+
+    Unhandled Exception Detection Algorithm
+
+    A subset of the aboved described callbacks is used to determine if an unhandled exception has occurred:
+    - ExceptionThrown: Record the originating FunctionID and the existance of the exception.
+    - ExceptionSearchCatcherFound: Record that the exception will be handled in a catching FunctionID.
+    - ExceptionUnwindFunctionEnter: If the exception was not handled (ExceptionSearchCatcherFound was not invoked),
+      then the exception is unhandled at this point. Do some extra bookkeeping in the case when the exception is
+      handled so that the exception is cleared once the frame with the corresponding catching FunctionID is encountered.
+
+    Current pitfalls:
+    - Algorithm doesn't account for exceptions thrown within exception filters. These will be seen as a new set of
+      callback invocations starting with throwing, searching, and unwinding, however, the callstack does not unwind
+      out of the exception filter and the original exception processing is resumed.
+    - Algorithm doesn't account for exceptions thrown within finally blocks. Presumably, these will be seen as new set
+      of callback invocations starting with throwing, searching, and unwinding. The original exception processing is
+      superceded by the new set of callbacks for the new exception.
+*/
+
+
+ExceptionTracker::ExceptionTracker(
+    const shared_ptr<ILogger>& logger,
+    const shared_ptr<ThreadDataManager> threadDataManager,
+    ICorProfilerInfo2* corProfilerInfo)
+{
+    _corProfilerInfo = corProfilerInfo;
+    _logger = logger;
+    _threadDataManager = threadDataManager;
+}
+
+void ExceptionTracker::AddProfilerEventMask(DWORD& eventsLow)
+{
+    eventsLow |= COR_PRF_MONITOR::COR_PRF_ENABLE_STACK_SNAPSHOT;
+}
+
+HRESULT ExceptionTracker::ExceptionThrown(ThreadID threadId, ObjectID objectId)
+{
+    // CAUTION: Do not store the exception ObjectID. It is not guaranteed to be correct
+    // outside of the ExceptionThrown callback without updating it using GC callbacks.
+
+    HRESULT hr = S_OK;
+
+    IfFailLogRet(_threadDataManager->SetHasException(threadId));
+
+    // Exception throwing is common; don't pay to calculate method name if it won't be logged.
+    if (_logger->IsEnabled(LogLevel::Debug))
+    {
+        FunctionID functionId = 0;
```

Comment:
Are we using this anymore? Or is it just a remnant from a prior iteration?

Reply:
Seems to be a remnant; I will remove in the next iteration